web/packages/teleport/src/services/integrations/integrations.ts

```diff
@@ -292,10 +293,24 @@ export const integrationService = {
       .then(resp => resp.serviceDashboardUrl);
   },
 
-  async createAwsAppAccess(integrationName): Promise<App> {
-    return api
-      .post(cfg.getAwsAppAccessUrl(integrationName), null)
-      .then(makeApp);
+  async createAwsAppAccess(
+    integrationName,
+    req: CreateAwsAppAccessRequest
+  ): Promise<App> {
+    // TODO(kimlisa): DELETE IN 19.0 - replaced by v2 endpoint.
+    if (!req.labels || !Object.keys(req.labels).length) {
```

Comment:
I think it would be better to create a `v2` `createAwsAppAccess` here so that `createAwsAppAccessV2` always hits the new API and transition in `v19` will be safe. Mostly because the `CreateAwsAppAccessRequest` has the optional label field regardless of `v1` and `v2` API which makes it a bit unpredictable on the implication of empty label field.